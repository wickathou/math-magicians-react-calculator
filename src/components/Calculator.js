import React from 'react';
import Button from './Button';
import Result from './Result';
import '../App.css';

function Calculator() {
  const regularButton = 'box-border border-2 border-neutral-400 col-span-1 bg-indigo-500 flex justify-center items-center'
  const wideButton = 'box-border border-2 border-neutral-400 col-span-2 bg-indigo-500 flex justify-center items-center'
  const display = 'box-border border-2 border-neutral-400 col-span-4 bg-indigo-500 flex justify-center items-center text-white'
  const actionColor = 'bg-orange-500'
  const numberColor = 'bg-stone-200'
  const displayColor = 'bg-neutral-800'
  const calculatorStyle = 'grid grid-cols-4 w-80 h-80 box-border border-4 border-neutral-400'

  return (
    <div className={calculatorStyle}>
      <Result className={display+' '+displayColor} result='Calculator'/>

      <Button className={regularButton+' '+numberColor} identifier='AC'/>
      <Button className={regularButton+' '+numberColor} identifier='+/-'/>
      <Button className={regularButton+' '+numberColor} identifier='%'/>
      <Button className={regularButton+' '+actionColor} identifier='/'/>

      <Button className={regularButton+' '+numberColor} identifier='7'/>
      <Button className={regularButton+' '+numberColor} identifier='8'/>
      <Button className={regularButton+' '+numberColor} identifier='9'/>
      <Button className={regularButton+' '+actionColor} identifier='x'/>

      <Button className={regularButton+' '+numberColor} identifier='4'/>
      <Button className={regularButton+' '+numberColor} identifier='5'/>
      <Button className={regularButton+' '+numberColor} identifier='6'/>
      <Button className={regularButton+' '+actionColor} identifier='-'/>

      <Button className={regularButton+' '+numberColor} identifier='1'/>
      <Button className={regularButton+' '+numberColor} identifier='2'/>
      <Button className={regularButton+' '+numberColor} identifier='3'/>
      <Button className={regularButton+' '+actionColor} identifier='+'/>

      <Button className={wideButton+' '+numberColor} identifier='0'/>
      <Button className={regularButton+' '+numberColor} identifier='.'/>
      <Button className={regularButton+' '+actionColor} identifier='='/>
    </div>
  );
}

export default Calculator;