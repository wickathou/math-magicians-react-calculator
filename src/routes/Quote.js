import React from 'react';

function Quote() {
  return (
    <>
      <h2 className="font-bold text-lg">
        &quot;Pure mathematics is, in its way, the poetry of logical ideas.&quot;
      </h2>
      <h3 className="mt-4">
        Albert Einstein
      </h3>
    </>
  );
}

export default Quote;
