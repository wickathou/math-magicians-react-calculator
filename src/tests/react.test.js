import React from 'react';
import renderer from 'react-test-renderer';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import Home from '../routes/Home';
import Quote from '../routes/Quote';
import Calculator from '../routes/Calculator';
import '@testing-library/jest-dom';

describe('calculate tests for different button inputs', () => {
  it('renders Home element correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Quote element correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Calculator element correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Tests calculator display functionality', () => {
  it('renders Calculator and test if key pressed updates value of calculator display', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('1'));
    expect(screen.getByRole('dialog')).toHaveTextContent('1');
  });

  it('renders Calculator and test if number inputs are correcly displayed', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('1'));
    fireEvent.click(screen.getByRole('2'));
    fireEvent.click(screen.getByRole('3'));
    expect(screen.getByRole('dialog')).toHaveTextContent('123');
  });

  it('renders Calculator and test if key pressed updates value of calculator display', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('1'));
    fireEvent.click(screen.getByRole('+'));
    fireEvent.click(screen.getByRole('1'));
    fireEvent.click(screen.getByRole('='));
    expect(screen.getByRole('dialog')).toHaveTextContent('2');
  });
});
