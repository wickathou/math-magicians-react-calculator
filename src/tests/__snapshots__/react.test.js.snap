// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate tests for different button inputs renders Calculator element correctly 1`] = `
<div
  className="calculator-style"
>
  <div
    className="display-style"
  >
    <h2
      role="dialog"
    >
      0
    </h2>
  </div>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="AC"
    >
      AC
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="+/-"
    >
      +/-
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="%"
    >
      %
    </h2>
  </button>
  <button
    className="act-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="รท"
    >
      รท
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="7"
    >
      7
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="8"
    >
      8
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="9"
    >
      9
    </h2>
  </button>
  <button
    className="act-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="x"
    >
      x
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="4"
    >
      4
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="5"
    >
      5
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="6"
    >
      6
    </h2>
  </button>
  <button
    className="act-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="-"
    >
      -
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="1"
    >
      1
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="2"
    >
      2
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="3"
    >
      3
    </h2>
  </button>
  <button
    className="act-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="+"
    >
      +
    </h2>
  </button>
  <button
    className="wide-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="0"
    >
      0
    </h2>
  </button>
  <button
    className="reg-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="."
    >
      .
    </h2>
  </button>
  <button
    className="act-btn"
    onClick={[Function]}
    type="button"
  >
    <h2
      role="="
    >
      =
    </h2>
  </button>
</div>
`;

exports[`calculate tests for different button inputs renders Home element correctly 1`] = `
<p
  className="text-2xl"
>
  This calculator app is a web-based application built using the popular React framework for creating user interfaces, Snowpack for faster and more efficient development, and Tailwind for easy and customizable styling. This calculator app allows users to perform basic arithmetic operations, including addition, subtraction, multiplication, and division, as well as calculate percentages and clear the current value. The interface is designed to be user-friendly and responsive, allowing for easy input and output on a range of devices. With the power of React, the app ensures fast and seamless user experience, while Snowpack and Tailwind make development and styling a breeze.
</p>
`;

exports[`calculate tests for different button inputs renders Quote element correctly 1`] = `
[
  <h2
    className="font-bold text-lg"
  >
    "Pure mathematics is, in its way, the poetry of logical ideas."
  </h2>,
  <h3
    className="mt-4"
  >
    Albert Einstein
  </h3>,
]
`;
